File: app/src/main/java/soup/movie/ui/main/settings/SettingsContract.java
package soup.movie.ui.main.settings;

import android.support.annotation.NonNull;

import soup.movie.ui.BaseContract;

public interface SettingsContract {

    interface Presenter extends BaseContract.Presenter<View> {
    }

    interface View extends BaseContract.View {

        void render(@NonNull SettingsViewState viewState);
    }
}

File: app/src/main/java/soup/movie/ui/main/settings/SettingsContract.kt
package soup.movie.ui.main.settings

import soup.movie.ui.LegacyBaseContract

interface SettingsContract {

    interface Presenter : LegacyBaseContract.Presenter<View>

    interface View : LegacyBaseContract.View {

        fun render(viewState: SettingsViewState.TheaterListViewState)

        fun render(viewState: SettingsViewState.VersionViewState)
    }
}
